src/game_interpreter.cpp

```diff
@@ -1139,19 +1126,20 @@ std::vector<Game_Actor*> Game_Interpreter::GetActors(int mode, int id) {
 	return actors;
 }
 
-Game_Character* Game_Interpreter::GetCharacter(int character_id) const {
-	if (!event_info.page && character_id == Game_Character::CharThisEvent) {
+Game_Character* Game_Interpreter::GetCharacter(int event_id) const {
+	if (event_id == Game_Character::CharThisEvent) {
+		event_id = GetThisEventId();
 		// Is a common event
 		if (event_id == 0) {
 			// With no map parent
-			Output::Warning("Can't use ThisEvent in common event %d: Not called from a map event", event_info.x);
+			Output::Warning("Can't use ThisEvent in common event: Not called from a map event");
```

Comment:
is there no way anymore to get the common event id for better debug output?

Reply:
I took at a look at this and it's not easy. Mostly because we would have to augment the call stack to store the id for each stack frame. I could easily store the id of the running event but I wouldn't be able to fix it when I restore it.

This would need some refactor to not use LSD data directory but implement the call stack in a similar way to keyinput with translation to/from LSD data.

I agree we should have a good interpreter debugging mechanism, with first class logging and other things.

Can we make an issue and punt on that for now? Don't want to add another complex commit to this which will rebase hell everything.